cluster-lifecycle/update-managedcluster-status-to-include-clusterinfo.md

```diff
@@ -0,0 +1,256 @@
+---
+title: update-managedcluster-status-to-include-clusterinfo
+authors:
+  - "@elgnay"
+reviewers:
+  - "@qiujian16"
+approvers:
+  - "@deads2k"
+  - "@pmorie"
+  - "@mdelder"
+creation-date: 2020-10-30
+last-updated: 2020-11-03
+status: provisional
+---
+
+# Update ManagedCluster status to include ClusterInfo
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+The proposed work would integrate the ClusterInfoStatus into ManagedCluster:
+
+https://github.com/open-cluster-management/multicloud-operators-foundation/blob/master/pkg/apis/internal.open-cluster-management.io/v1beta1/clusterinfo_types.go#L22-L62
+
+## Motivation
+
+Today, several label properties, including clusterID, cloud and vendor, are updated on ManagedCluster objects that can be modified by users, creating invalid data for matching behavior.
+
+As discussed in https://docs.google.com/document/d/1Mb7Z7UEYabLGAhFpcxzD9ymEdKd4rFFj0H-WS4skY2Q/edit and on the Architecture Forum on Sept 17, it would be a good idea to include a way to validate whether the labels are correct by updating the StatusConditions on ManagedCluster with information provided by the Klusterlet. Today, ManagedClusterInfo captures this information in a separate API object.
+
+The way provided should be extensible and neutral, and can work with different implementations.
```

Comment:
Let's say like

It is required from other components that more cluster specific info should be add into  managed cluster API. These informations includes the clusterid, the ocp distribution version, vendor and cloud provider etc. On another hand, managedcluster API should be designed to have a generic and extensible structure to include these info that also can work with different implementations.

Reply:
Updated.